paf-mvp-client-express/src/client-node.ts

```diff
@@ -67,417 +56,420 @@ export interface ClientNodeConfig extends Config {
+      this.endSpan(jsonProxyEndpoints.verifyRead)
     );
 
-    app.expressApp.post(
+    this.app.expressApp.post(
       jsonProxyEndpoints.signPrefs,
-      cors(corsOptions),
-      checkOrigin(jsonProxyEndpoints.signPrefs),
-      (req, res) => {
-        logger.Info(jsonProxyEndpoints.signPrefs);
-        try {
-          const { identifiers, unsignedPreferences } = getPayload<PostSignPreferencesRequest>(req);
-          res.json(client.buildPreferences(identifiers, unsignedPreferences.data));
-        } catch (e) {
-          logger.Error(jsonProxyEndpoints.signPrefs, e);
-          // FIXME finer error return
-          const error: ClientNodeError = {
-            type: ClientNodeErrorType.UNKNOWN_ERROR,
-            details: '',
-          };
-          res.status(400);
-          res.json(error);
-        }
-      }
+      this.originValidator.cors,
+      this.originValidator.checkOrigin(jsonProxyEndpoints.signPrefs),
+      this.startSpan(jsonProxyEndpoints.signPrefs),
+      this.signPreferences.bind(this),
+      this.handleErrors(jsonProxyEndpoints.signPrefs),
+      this.endSpan(jsonProxyEndpoints.signPrefs)
     );
 
     // *****************************************************************************************************************
     // ***************************************************************************************************** JSON - SEED
     // *****************************************************************************************************************
-
-    app.expressApp.post(
+    this.app.expressApp.post(
       jsonProxyEndpoints.createSeed,
-      cors(corsOptions),
-      checkOrigin(jsonProxyEndpoints.createSeed),
-      (req, res) => {
-        logger.Info(jsonProxyEndpoints.createSeed);
-        try {
-          const request = JSON.parse(req.body as string) as PostSeedRequest;
-          const seed = client.buildSeed(request.transaction_ids, request.data);
-          const response = seed as PostSeedResponse; // For now, the response is only a Seed.
-          res.json(response);
-        } catch (e) {
-          logger.Error(jsonProxyEndpoints.createSeed, e);
-          // FIXME finer error return
-          const error: ClientNodeError = {
-            type: ClientNodeErrorType.UNKNOWN_ERROR,
-            details: '',
-          };
-          res.status(400);
-          res.json(error);
-        }
-      }
+      this.originValidator.cors,
+      this.originValidator.checkOrigin(jsonProxyEndpoints.createSeed),
+      this.startSpan(jsonProxyEndpoints.createSeed),
+      this.createSeed.bind(this),
+      this.handleErrors(jsonProxyEndpoints.createSeed),
+      this.endSpan(jsonProxyEndpoints.createSeed)
     );
   }
 
+  restRead(req: Request, res: Response, next: NextFunction) {
+    try {
+      const url = this.client.getReadRestUrl(req);
+      res.send(url.toString());
+      next();
+    } catch (e) {
+      const error: ClientNodeError = {
```

Comment:
Do we need to try cath in every `rest` methods?

The custom error handler should be able to retrieve the throwed error and set the status and the json accordingly.

Reply:
Handled in a future PR.

Reply:
Ok.